frontend/src/crosssection/components/CrossMap.vue

```diff
@@ -0,0 +1,199 @@
+  DATABC_ROADS_LAYER,
+  DATABC_CADASTREL_LAYER,
+  wellsBaseAndArtesianLayer,
+  WELLS_BASE_AND_ARTESIAN_LAYER_ID,
+  FOCUSED_WELLS_SOURCE_ID,
+  focusedWellsLayer,
+  FOCUSED_WELL_IMAGE_ID,
+  FOCUSED_WELL_ARTESIAN_IMAGE_ID,
+  FOCUSED_WELL_CLOSED_IMAGE_ID,
+  wellLayerFilter,
+  WELLS_SOURCE
+} from '../../common/mapbox/layers'
+import { LegendControl } from '../../common/mapbox/controls'
+import { createWellPopupElement } from '../../common/mapbox/popup'
+import { PulsingWellImage, PulsingArtesianWellImage, PulsingClosedWellImage } from '../../common/mapbox/images'
+import {CENTRE_LNG_LAT_BC, buildWellsGeoJSON } from '../../common/mapbox/geometry'
+
+import wellsAllLegendSrc from '../../common/assets/images/wells-all.svg'
+import wellsArtesianLegendSrc from '../../common/assets/images/wells-artesian.svg'
+import wellsClosedLegendSrc from '../../common/assets/images/wells-closed.svg'
+import { setupFeatureTooltips } from '../../common/mapbox/popup'
+
+const WELL_FEATURE_PROPERTIES_FOR_POPUP = [
+  'well_tag_number',
+  'identification_plate_number',
+  'street_address',
+  'is_published'
+]
+const FOCUSED_WELL_PROPERTIES = WELL_FEATURE_PROPERTIES_FOR_POPUP.concat(['artesian_conditions', 'well_status'])
+
+export default {
+  name: 'CrossMap',
+  props: [
+    'focusedWells'
+  ],
+  data () {
+    return {
+      map: null,
+      browserUnsupported: false,
+      mapLayers: [
+        {
+          show: true,
+          id: WELLS_BASE_AND_ARTESIAN_LAYER_ID,
+          label: 'Wells',
+          legend: [
+            {
+              imageSrc: wellsAllLegendSrc,
+              label: 'all'
+            },
+            {
+              imageSrc: wellsArtesianLegendSrc,
+              label: 'artesian'
+            },
+            {
+              imageSrc: wellsClosedLegendSrc,
+              label: 'closed/abandoned'
+            }
+          ]
+        }
+      ]
+    }
+  },
+  mounted () {
+    this.$emit('mapLoading')
+
+    this.initMapBox()
+  },
+  destroyed () {
+    this.map.remove()
+    this.map = null
+  },
+  methods: {
+    initMapBox () {
+      if (!mapboxgl.supported()) {
+        this.browserUnsupported = true
+        return
+      }
+
+      var mapConfig = {
+        container: this.$el,
+        zoom: 4,
+        minZoom: 4,
+        maxPitch: 0,
+        dragRotate: false,
+        center: CENTRE_LNG_LAT_BC,
+        style: {
+          version: 8,
+          sources: {
+            [DATABC_ROADS_SOURCE_ID]: DATABC_ROADS_SOURCE,
+            [DATABC_CADASTREL_SOURCE_ID]: DATABC_CADASTREL_SOURCE,
+            [WELLS_SOURCE_ID]: WELLS_SOURCE,
+            [FOCUSED_WELLS_SOURCE_ID]: { type: 'geojson', data: buildWellsGeoJSON([]) }
+          },
+          layers: [
+            DATABC_ROADS_LAYER,
+            DATABC_CADASTREL_LAYER,
+            //should we filter based on user role? this page is already behind keycloak
+            wellsBaseAndArtesianLayer({ filter: wellLayerFilter(true) }),
```

Comment:
Yes, this should take in a showUnpublished variable. Once we move things to the BE, we can remove all these filters.

Reply:
now filters on unpublished var